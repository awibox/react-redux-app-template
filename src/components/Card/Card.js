import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
// Styles
import styles from './Card.scss';

const Card = (props) => (
  <div className={classNames(styles.card, { [props.className]: props.className })}>
    {props.children}
  </div>
);

Card.propTypes = {
  children: PropTypes.any,
  className: PropTypes.string,
};

export default Card;
