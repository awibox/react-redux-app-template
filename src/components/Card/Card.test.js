import React from 'react';
import Card from './Card';
import { shallow } from 'enzyme';

const testClassName = 'test-class';

describe('Card', () => {
  it('should render correctly', () => {
    const component = shallow(<Card>
      <div>
        <div>
          Test Card
        </div>
      </div>
    </Card>);
    expect(component).toMatchSnapshot();
  });
  it('should have className', () => {
    const component = shallow(<Card className={testClassName}>Test button</Card>);
    expect(component.find('.' + testClassName)).toExist();
  });
});
