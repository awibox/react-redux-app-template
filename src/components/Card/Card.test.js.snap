// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render correctly 1`] = `
<div
  className="card"
>
  <div>
    <div>
      Test Card
    </div>
  </div>
</div>
`;
